src/Maestro/maestro-angular/src/app/page/build-graph-table/build-graph-table.component.ts

```diff
@@ -96,11 +99,47 @@ function sortBuilds(graph: BuildGraph): BuildData[] {
   result = result.reverse();
   for (const node of result) {
     node.isToolset = isToolset(node.build, graph);
+    node.hasIncoherentDependencies = hasIncoherentDependencies(node.build, result, false);
+    node.hasIncoherentDependenciesIncludingToolsets = hasIncoherentDependencies(node.build, result, true);
   }
 
   return result;
 }
 
+function hasIncoherentDependencies(build: Build, buildData: BuildData[], includeToolsets: boolean): boolean {
+  let y = buildData.find(x => x.build.id == build.id);
+  if(y!.hasIncoherentDependencies)
+  {
+    return true;
+  }
+
+  if (y!.build.dependencies) {
+    for (const dep of y!.build.dependencies) {
+      if(dep && dep!.isProduct){
+        let z = buildData.find(r => r.build.id == dep.buildId);
+
+        if((((!z!.coherent.withProduct && !includeToolsets) || (!z!.coherent.withAll && includeToolsets)) || 
+          ((z!.hasIncoherentDependencies && !includeToolsets) || (z!.hasIncoherentDependenciesIncludingToolsets && includeToolsets))) && 
+          z!.build.dependencies!.length == 0)
```

Comment:
>dependencies! [](start = 19, length = 13)

Why are we able to assert this isn't null here? We didn't check it anywhere...  Isn't this a null-ref potential?

Reply:
Was able to handle this differently after refactoring this bit of code.